exopy_hqc_legacy/tasks/tasks/instr/apply_mag_field_task.py

```diff
@@ -59,41 +59,47 @@ def perform(self, target_value=None):
 
         driver = self.driver
         normal_end = True
-        if (abs(driver.read_persistent_field() - target_value) >
-                driver.output_fluctuations):
+        # Only if driver.heater_state == 'Off' otherwise we wait for 
+        # post_switch_wait no matter what
+        if driver.heater_state == 'Off':
             job = driver.sweep_to_persistent_field()
             if job.wait_for_completion(self.check_for_interruption,
-                                       timeout=60, refresh_time=1):
+                                       timeout=60, refresh_time=3):
                 driver.heater_state = 'On'
                 sleep(self.post_switch_wait)
             else:
                 return False
 
+        if (abs(driver.read_persistent_field() - target_value) > 
+            driver.output_fluctuations):
             # set the magnetic field
             job = driver.sweep_to_field(target_value, self.rate)
             normal_end = job.wait_for_completion(self.check_for_interruption,
                                                  timeout=60,
                                                  refresh_time=10)
+            print('field:', driver.read_persistent_field())
 
         # Always close the switch heater when the ramp was interrupted.
         if not normal_end:
-            job.cancel()
-            driver.heater_state = 'Off'
-            sleep(self.post_switch_wait)
+            job.cancel() # stops the sweep and turn off the switch heater
```

Comment:
I don't see why cancelling the job should always close the switch heater. Also you only modified the cg4 and cs4 but I believe the Oxford source behaves differently.

Reply:
I didn't add this but it seems rather reasonable to me since the switch heater increases the He consumption

The oxford source is broken so I have not updated the driver (since I am not able to test it and I didn't want to leave something buggy behind)

Reply:
I am fine with closing the heater my concern is the same above regarding the location of that code inside cancel.